src/Elements/Serialization/ModelConverter.cs

```diff
@@ -98,7 +100,9 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
                         new ProfileToIdConverter(model.Profiles),
                         new SolidConverter(model.Materials)
                     },
-                NullValueHandling = NullValueHandling.Ignore
+                NullValueHandling = NullValueHandling.Ignore,
+                TypeNameHandling = TypeNameHandling.Auto,
```

Comment:
I don't think we want to add this. `TypeNameHandling` will add a field `$type` to all serialized objects, with the fully namespaced type that is the target for deserialization. We've already got a version of this with our `type` field. 

Is there a reason you need it?

Reply:
first, sorry, I broke this pull request trying to clean.  let's continue the convo here.
I have needed it to deserialize elements with properties where the property type is an interface, not specific classes.  This appears in my lastest versions of creating/using `IConnect` interfaces, for example.
```c#
    public interface IOneToManyConnection
    {
        IConnector Start{get;}
        IConnector[] End{get;}
    }
```
When one tries to deserialize a class that implements this interface, the Start and End properties throw errors, because whatever is stored there cannot have it's type identified.
I see now that the element type property seems like it SHOULD work... 
How about this for proceeding, are you ready to include those interfaces in Hypar.Elements?  I have some slight tweaks but I could make a different pull request that contained the serialization settings, the interfaces and some tests...  that sound like a plan, or do you want to hold off on putting those interfaces in the library?

Reply:
Just double checked, and when I serialize/deserialize the first sample of json does not work, but the second does.  It's an MEPConnector object stored in a `Dictionary<string, IConnector>`  I see that a type is written in both cases, but only when the $type property is there does the deserialization work... thoughts?
```json
 "Room Air": {
        "id": 23,
        "type": "mister.core.systems.mepconnector",
        "Direction": 0,
        "Medium": {
          "UniqueName": "Room Air"
        },
```
```json
 "Room Air": {
        "$type": "Mister.Core.Systems.MEPConnector, Mister.Core",
        "id": 23,
        "type": "mister.core.systems.mepconnector",
        "Direction": 0,
        "Medium": {
          "$type": "Mister.Core.Systems.Medium, Mister.Core",
          "UniqueName": "Room Air"
        },
```

Reply:
@ikeough   Just want to make sure you see this.  Not sure if notifications happen with closed pull requests.